lib/elixir/src/elixir_with.erl

```diff
@@ -1,7 +1,7 @@
 -module(elixir_with).
 -export([expand/3, translate/3]).
 -include("elixir.hrl").
-
+-define(ref(Meta), {ref, Meta, elixir_with}).
```

Comment:
> nil is the user space, than it is possible to fake

Having `nil` instead of `elixir_with` it is not possible to fake this way, at this step we rather doing rewriting, `ref` variable translation will be later and it will receive a counter to make it unique.
But in this context `elixir_with` makes `ref` variable to not leak.